src/ir/datalog/attribute.h

```diff
@@ -0,0 +1,64 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#ifndef SRC_IR_DATALOG_ATTRIBUTE_H_
+#define SRC_IR_DATALOG_ATTRIBUTE_H_
+
+#include <memory>
+
+#include "absl/strings/string_view.h"
+#include "src/ir/datalog/value.h"
+
+namespace raksha::ir::datalog {
+
+inline constexpr absl::string_view kStringAttributePayloadName =
+    "StringAttributePayload";
+inline constexpr absl::string_view kNumberAttributePayloadName =
+    "NumberAttributePayload";
+
+class AttributePayload : public Adt {
+ public:
+  using Adt::Adt;
+};
```

Comment:
`using AttributePayload = Adt` ?

Reply:
That would work, but I would lose a type constraint that way: anytime I annotate a `Record` or `Adt`'s type parameters as wanting an `AttributePayload`, I want only ADT branches that belong to the `AttributePayload` ADT. By making `AttributePayload` a subclass of `Adt`, the type system prevents ADT branches belong to other ADTs from being passed in. If I were to make `AttributePayload` a `using` alias of `Adt`, the type system would not protect me in that way.